activerecord/lib/active_record/errors.rb

```diff
@@ -182,7 +182,12 @@ def initialize(record = nil)
   # Wraps the underlying database error as +cause+.
   class StatementInvalid < AdapterError
     def initialize(message = nil, sql: nil, binds: nil, connection_pool: nil)
-      super(message || $!&.message, connection_pool: connection_pool)
+      message ||= $!&.message
+      if connection_pool && message
+        message = "Database configuration: #{connection_pool.db_config.name}\n#{message}"
```

Comment:
I don't think this is enough, or correct to only use the config name. The connection specification name is the class name that established the connection (in the example that would be `ActiveRecord::Base`), not the name of the `db_config` in the database.yml. So if we are going to include this it needs to mention all the relevant information (connection name, role, and config name).

Reply:
Thanks, included that information too. Probably we can improve the text of the message.